src/components/AvatarCropModal/AvatarCropModal.js

```diff
@@ -0,0 +1,312 @@
+    const initializeSliderContainer = useCallback(event => setSliderContainerSize(event.nativeEvent.layout.width), []);
+
+    /**
+     * Changes the modal state values to initial
+    */
+    const initializeImage = useCallback(() => {
+        translateY.value = 0;
+        translateX.value = 0;
+        scale.value = CONST.AVATAR_CROP_MODAL.MIN_SCALE;
+        rotation.value = 0;
+        translateSlider.value = 0;
+    }, []);
+
+    useEffect(() => {
+        if (!props.imageUri) {
+            return;
+        }
+        initializeImage();
+        Image.getSize(props.imageUri, (width, height) => {
+            // We need to have image sizes in shared values to properly calculate animation
+            originalImageHeight.value = height;
+            originalImageWidth.value = width;
+
+            // Slider and image rotation values are not updated, we need to change them slightly
+            // initializing image sizes, to display correctly
+            translateSlider.value += 0.01;
+            rotation.value += 360;
+        });
+    }, [props.imageUri, initializeImage]);
+
+    /**
+     * Validates that value is within the provided mix/max range.
+     *
+     * @param {Number} value
+     * @param {Array} minMax
+     * @returns {Number}
+    */
+    const clamp = useWorkletCallback((value, [min, max]) => interpolate(value, [min, max], [min, max], 'clamp'), []);
+
+    /**
+     * Returns current image size taking into account scale and rotation.
+     *
+     * @returns {Object}
+    */
+    const getDisplayedImageSize = useWorkletCallback(() => {
+        let height = imageContainerSize * scale.value;
+        let width = imageContainerSize * scale.value;
+
+        // Since the smaller side will be always equal to the imageContainerSize multiplied by scale,
+        // another side can be calculated with aspect ratio.
+        if (originalImageWidth.value > originalImageHeight.value) {
+            width *= originalImageWidth.value / originalImageHeight.value;
+        } else {
+            height *= originalImageHeight.value / originalImageWidth.value;
+        }
+
+        return {height, width};
+    }, [imageContainerSize, scale]);
+
+    /**
+     * Validates the offset to prevent overflow, and updates the image offset.
+     *
+     * @param {Number} newX
+     * @param {Number} newY
+    */
+    const updateImageOffset = useWorkletCallback((offsetX, offsetY) => {
+        const {height, width} = getDisplayedImageSize();
+        const maxOffsetX = (width - imageContainerSize) / 2;
+        const maxOffsetY = (height - imageContainerSize) / 2;
+        translateX.value = clamp(offsetX, [maxOffsetX * -1, maxOffsetX]);
+        translateY.value = clamp(offsetY, [maxOffsetY * -1, maxOffsetY]);
+    }, [imageContainerSize, scale, clamp]);
+
+    /**
+     * Calculates new x & y image translate value on image panning
+     * and updates image's offset.
```

Comment:
This should also have params added to jsdoc

Reply:
Why? This hook is from the react-native-reanimated library. It is **not custom**. Also, we are not calling result of this hook directly from our code base. We just passing it to the react-native-gesture-handler (another library from reanimated)